db/migrate/20180821112856_create_openshift_service_catalog_tables.rb

```diff
@@ -0,0 +1,114 @@
+class CreateOpenshiftServiceCatalogTables < ActiveRecord::Migration[5.0]
+  def change
+    create_table :container_service_brokers, :id => :bigserial, :force => :cascade do |t|
+      t.string :name
+      t.string :kind
+      t.string :ems_ref
+      t.integer :resource_version
+      t.text :url
+
+      t.references :ems_id, :type => :bigint, :index => true, :references => :ext_management_systems
+      t.references :container_projects, :type => :bigint, :index => true
+
+      t.jsonb :extra
+
+      t.datetime :ems_created_on
+      t.timestamps
+    end
+
+    create_table :container_service_classes, :id => :bigserial, :force => :cascade do |t|
+      t.string :name
+      t.string :kind
+      t.string :status
+      t.string :ems_ref
+      t.integer :resource_version
+      t.text :description
+      t.boolean :bindable
+      t.boolean :plan_updatable
+
+      t.references :ems_id, :type => :bigint, :index => true, :references => :ext_management_systems
+      t.references :container_projects, :type => :bigint, :index => true
+      t.references :container_service_brokers, :type => :bigint, :index => true
+
+      t.jsonb :extra
+
+      t.datetime :ems_created_on
+      t.timestamps
+    end
+
+    create_table :container_service_plans, :id => :bigserial, :force => :cascade do |t|
+      t.string :name
+      t.string :kind
+      t.string :status
+      t.string :ems_ref
+      t.integer :resource_version
+      t.text :description
+      t.boolean :free
+
+      t.references :ems_id, :type => :bigint, :index => true, :references => :ext_management_systems
+      t.references :container_projects, :type => :bigint, :index => true
+      t.references :container_service_classes, :type => :bigint, :index => true
+
+      t.jsonb :extra
+
+      t.datetime :ems_created_on
+      t.timestamps
+    end
+
+    create_table :container_service_instances, :id => :bigserial, :force => :cascade do |t|
+      t.string :name
+      t.string :kind
+      t.string :status
+      t.string :ems_ref
+      t.string :secret_name
+      t.integer :resource_version
+
+      t.references :ems_id, :type => :bigint, :index => true, :references => :ext_management_systems
+      t.references :container_projects, :type => :bigint, :index => true
+      t.references :container_service_classes, :type => :bigint, :index => {:name => 'csi_on_container_service_classes'}
+      t.references :container_service_plans, :type => :bigint, :index => true
+
+      t.jsonb :parameters
+      t.jsonb :parameters_from
+      t.jsonb :extra
```

Comment:
@Ladas Would the extra column contain the status information about a service which is a nested hash. Here is an example of a failed service
```
apiVersion: v1
items:
- apiVersion: servicecatalog.k8s.io/v1beta1
  kind: ServiceInstance
  metadata:
    creationTimestamp: 2018-08-21T14:02:39Z
    finalizers:
    - kubernetes-incubator/service-catalog
    generateName: mariadb-persistent-
    generation: 1
    name: mariadb-persistent-qdkzt
    namespace: default
    resourceVersion: "35703842"
    selfLink: /apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances/mariadb-persistent-qdkzt
    uid: dd871fef-a54a-11e8-8fb9-0a580a800009
  spec:
    clusterServiceClassExternalName: mariadb-persistent
    clusterServiceClassRef:
      name: 1750b5d6-0d11-11e8-906a-d094660d31fb
    clusterServicePlanExternalName: default
    clusterServicePlanRef:
      name: 1750b5d6-0d11-11e8-906a-d094660d31fb
    externalID: 76af97e3-5650-4583-ae85-27294677f88d
    parametersFrom:
    - secretKeyRef:
        key: parameters
        name: mariadb-persistent-parametersbori0
    updateRequests: 0
    userInfo:
      extra:
        scopes.authorization.openshift.io:
        - user:full
      groups:
      - system:authenticated:oauth
      - system:authenticated
      uid: ""
      username: admin
  **status:**
    asyncOpInProgress: false
    conditions:
    - lastTransitionTime: 2018-08-21T14:02:44Z
      message: 'The provision call failed and will be retried: Error communicating
        with broker for provisioning: Put https://apiserver.openshift-template-service-broker.svc:443/brokers/template.openshift.io/v2/service_instances/76af97e3-5650-4583-ae85-27294677f88d?accepts_incomplete=true:
        dial tcp: lookup apiserver.openshift-template-service-broker.svc on [::1]:53:
        read udp [::1]:36349->[::1]:53: read: connection refused'
      reason: ErrorCallingProvision
      status: "False"
      type: Ready
    currentOperation: Provision
    deprovisionStatus: Required
    inProgressProperties:
      clusterServicePlanExternalID: 1750b5d6-0d11-11e8-906a-d094660d31fb
      clusterServicePlanExternalName: default
      parameterChecksum: 71118b9c659a984212608999d0323deda723b5cdba0805a0e350fc49931b6604
      parameters:
        DATABASE_SERVICE_NAME: <redacted>
        MARIADB_VERSION: <redacted>
        MEMORY_LIMIT: <redacted>
        MYSQL_DATABASE: <redacted>
        NAMESPACE: <redacted>
        VOLUME_CAPACITY: <redacted>
      userInfo:
        extra:
          scopes.authorization.openshift.io:
          - user:full
        groups:
        - system:authenticated:oauth
        - system:authenticated
        uid: ""
        username: admin
    operationStartTime: 2018-08-21T14:02:44Z
    orphanMitigationInProgress: false
    reconciledGeneration: 0
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

```

Reply:
yea we can put it to extra or do separate jsob column for it. The extra is for whatever else we need and we do not have column for (for start, we won't be able to order and search using the stuff in jsonb, since we don't have that coded in)

Reply:
@syncrou @mkanoor but given the big amount of hash data, it would be nice if you guys would tell me what we actually need. So we don't bloat our DB with data we are not using for anything

Reply:
@Ladas - At this time I'm not fully sure if we know 100% what we need.  I'm not opposed to jamming the full hash into the extra column for now so we have it, with the understanding that we'll adjust it and add columns once we're set on what we need.  @mkanoor Thoughts?